rgd/geodata/models/fmv/etl.py

```diff
@@ -17,59 +16,44 @@
-    try:
-        client = docker.from_env(version='auto', timeout=3600)
-        _ = client.images.pull(image_name)
-
-        with video_file.yield_local_path() as dataset_path:
-            logger.info('Running dump-klv with data %s' % (dataset_path))
-            tmpdir = tempfile.mkdtemp()
-            output_path = os.path.join(tmpdir, os.path.basename(video_file.file.name) + '.klv')
-            stderr_path = os.path.join(tmpdir, 'stderr.dat')
-            cmd = [
-                'docker',
-                'run',
-                '--rm',
-                '-i',
-                image_name,
-            ]
-            logger.info(
-                'Running %s <%s >%s 2>%s'
-                % (
-                    ' '.join([shlex.quote(c) for c in cmd]),
-                    shlex.quote(str(dataset_path)),
-                    shlex.quote(output_path),
-                    shlex.quote(stderr_path),
-                )
-            )
-            try:
-                subprocess.check_call(
-                    cmd,
-                    stdin=open(dataset_path, 'rb'),
-                    stdout=open(output_path, 'wb'),
-                    stderr=open(stderr_path, 'wb'),
-                )
-                result = 0
-            except subprocess.CalledProcessError as exc:
-                result = exc.returncode
-                logger.info('Failed to successfully run image (%r)' % (exc))
-                raise exc
-            logger.info('Finished running image with result %r' % result)
-            # Store result
-            fmv_file_entry.klv_file.save(os.path.basename(output_path), open(output_path, 'rb'))
-            fmv_file_entry.save(
-                update_fields=[
-                    'klv_file',
-                ]
+    with video_file.yield_local_path() as dataset_path:
+        logger.info('Running dump-klv with data %s' % (dataset_path))
+        tmpdir = tempfile.mkdtemp()
+        output_path = os.path.join(tmpdir, os.path.basename(video_file.file.name) + '.klv')
+        stderr_path = os.path.join(tmpdir, 'stderr.dat')
+        cmd = [
+            'kwiver',
+            'dump-klv',
+            dataset_path,
+        ]
+        try:
+            subprocess.check_call(
+                cmd,
+                stdin=open(dataset_path, 'rb'),
+                stdout=open(output_path, 'wb'),
+                stderr=open(stderr_path, 'wb'),
             )
-            shutil.rmtree(tmpdir)
-    except Exception as exc:
-        logger.exception('Internal error running dump-klv')
-        raise exc
+            result = 0
+        except subprocess.CalledProcessError as exc:
+            result = exc.returncode
+            logger.info('Failed to successfully run dump-klv (%r)' % (exc))
+            raise exc
+        logger.info('Finished running dump-klv with result %r' % result)
+        # Store result
+        fmv_file_entry.klv_file.save(os.path.basename(output_path), open(output_path, 'rb'))
```

Comment:
It looks like this file descriptor isn't getting closed. Let's use a context manager.

Reply:
We do this in a number of places... I will go through and clean up all of these